### PR TITLE
JDK-8301506: Replace NULL with nullptr in os_cpu/linux_ppc

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory os_cpu/linux_ppc. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301506](https://bugs.openjdk.org/browse/JDK-8301506): Replace NULL with nullptr in os_cpu/linux_ppc

### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Richard Reingruber](https://openjdk.org/census#rrich) (@reinrich - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12335/head:pull/12335` \
`$ git checkout pull/12335`

Update a local copy of the PR: \
`$ git checkout pull/12335` \
`$ git pull https://git.openjdk.org/jdk pull/12335/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12335`

View PR using the GUI difftool: \
`$ git pr show -t 12335`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12335.diff">https://git.openjdk.org/jdk/pull/12335.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os_cpu/linux_ppc/javaThread_linux_ppc.cpp [hunk 1]
```diff
@@ -60,17 +60,17 @@ bool JavaThread::pd_get_top_frame_for_profiling(frame* fr_addr, void* ucontext,
...
     }
 
     frame ret_frame((intptr_t*)uc->uc_mcontext.regs->gpr[1/*REG_SP*/], pc);
 
-    if (ret_frame.fp() == NULL) {
+    if (ret_frame.fp() == nullptr) {
       // The found frame does not have a valid frame pointer.
       // Bail out because this will create big trouble later on, either
-      //  - when using istate, calculated as (NULL - ijava_state_size) or
+      //  - when using istate, calculated as (null - ijava_state_size) or
```
Comment (jdksjolen) [inline src/hotspot/os_cpu/linux_ppc/javaThread_linux_ppc.cpp:73]: nullptr

**Review by jdksjolen — COMMENTED** [review]
One small fix.

**Review by kimbarrett — APPROVED** [review]
Looks good, and trivial.

**Review by reinrich — APPROVED** [review]
Looks good.
Thanks, Richard.

**Comment by jdksjolen** [comment]
Thanks!

/integrate

**Pull request closed without merging**